docs/typescript.md

```diff
@@ -90,26 +104,26 @@ You can type the props of styled components.
 import styled from 'react-emotion'
 
 type ImageProps = {
-  src: string,
+  src: string;
   width: number;
 }
 
-const Image0 = styled('div')`
-  width: ${(props: ImageProps) => props.width};
+const Image0 = styled<ImageProps, 'div'>('div')`
+  width: ${(props) => props.width};
   background: url(${(props: ImageProps) => props.src}) center center;
   background-size: contain;
 `
 
 // Or with object styles
 
-const Image1 = styled('div')({
+const Image1 = styled<ImageProps, 'div'>('div')({
```

Comment:
Could you explain why people would want to do this over the generic way below?

Reply:
@mitchellhamilton Actually, I don't think this is the preferable way.... I add this because of compatibilities. If you think 'Document only for compatibility reason is not needed', I will change this.

Reply:
However, in case of string style, as you already know, situation is little different, since we cannot use generic template string in TS < 2.9, and even in TS >= 2.9, `vscode-styled-components` does not support syntax highlight.

Reply:
Could you move it below the one that people should use and add a comment about it only being there for compatibility and that people shouldn’t write new code with it?

Reply:
OK, I will move this and add a comment.